orttraining/orttraining/python/training/ortmodule/_execution_agent.py

```diff
@@ -2,134 +2,327 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+import onnx
+from onnx import ModelProto
+import torch
+from torch.utils.dlpack import to_dlpack
+
 
 import onnxruntime
+from onnxruntime import SessionOptions
 from onnxruntime.capi import _pybind_state as C
-from onnxruntime.capi.onnxruntime_inference_collection import IOBinding, OrtValue
-from onnxruntime.capi._pybind_state import TrainingAgent as C_TrainingAgent
+from onnxruntime.capi.onnxruntime_inference_collection import IOBinding, OrtValue, get_ort_device_type
+from onnxruntime.capi._pybind_state import OrtValueVector, PartialGraphExecutionState, RunOptions, TrainingAgent as C_TrainingAgent
+from . import _utils
+from ._graph_execution_manager import RunStateInfo
```

Comment:
Does this not introduce a cyclic dependency? May we move ```RunStateInfo``` to a file of its own, or utils or here?

Reply:
No `_graph_execution_manager.py ` does not import this file